horovod/mxnet/__init__.py

```diff
@@ -148,5 +150,5 @@ def broadcast_parameters(params, root_rank=0):
         raise ValueError('invalid params of type: %s' % type(params))
 
     # Run broadcasts.
-    for i, tensor in enumerate(tensors):
-        broadcast_(tensor, root_rank, str(i))
+    for tensor, name in zip(tensors, names):
+        broadcast_(tensor, root_rank, name=str(name))
```

Comment:
Is name here always a string?

Reply:
This is related to my comment above, where `test_horovod_broadcast_grad` was using integer keys in the passed in params dictionary, so this conversion to string was necessary to pass that test. However, I think passing a dictionary with non-string keys to this method should actually be considered user error because the params dictionaries from `Module.get_params()` or `Block.collect_params()` have string keys right?